app/utils/chart-config/versioning.ts

```diff
@@ -0,0 +1,148 @@
+export const CHART_CONFIG_VERSION = "1.0.2";
+
+const migrations = [
+  // MAP
+  // baseLayer {
+  //   + locked
+  //   + bbox
+  // }
+  {
+    from: "1.0.0",
+    to: "1.0.1",
+    up: (config: any) => {
+      let newConfig = { ...config, version: "1.0.1" };
+
+      if (newConfig.chartType === "map") {
+        const { baseLayer } = newConfig;
+        newConfig = {
+          ...newConfig,
+          baseLayer: {
+            show: baseLayer.show,
+            locked: false,
+          },
+        };
+      }
+
+      return newConfig;
+    },
+    down: (config: any) => {
+      let newConfig = { ...config, version: "1.0.0" };
+
+      if (newConfig.chartType === "map") {
+        const { baseLayer } = newConfig;
+        newConfig = {
+          ...newConfig,
+          baseLayer: {
+            show: baseLayer.show,
+          },
+        };
+      }
+
+      return newConfig;
+    },
+  },
+  // MAP
+  // symbolLayer {
+  //   + colors {
+  //     + type
+  //     + value
+  //     + opacity
+  //   }
+  //   - color
+  // }
+  {
+    from: "1.0.1",
+    to: "1.0.2",
+    up: (config: any) => {
+      let newConfig = { ...config, version: "1.0.2" };
+
+      if (newConfig.chartType === "map") {
+        const { fields } = newConfig;
+        const { symbolLayer } = fields;
+        newConfig = {
+          ...newConfig,
+          fields: {
+            ...fields,
+            symbolLayer: {
+              show: symbolLayer.show,
+              componentIri: symbolLayer.componentIri,
+              measureIri: symbolLayer.measureIri,
+              colors: {
+                type: "fixed",
+                value: symbolLayer.color,
+                opacity: 80,
+              },
+            },
+          },
+        };
+      }
+
+      return newConfig;
+    },
+    down: (config: any) => {
+      let newConfig = { ...config, version: "1.0.1" };
+
+      if (newConfig.chartType === "map") {
+        const { fields } = newConfig;
+        const { symbolLayer } = fields;
+        newConfig = {
+          ...newConfig,
+          fields: {
+            ...fields,
+            symbolLayer: {
+              show: symbolLayer.show,
+              componentIri: symbolLayer.componentIri,
+              measureIri: symbolLayer.measureIri,
+              color: symbolLayer.colors.value,
+            },
+          },
+        };
+      }
+
+      return newConfig;
+    },
+  },
+];
+
+export const migrateChartConfig = (
+  config: any,
+  toVersion: string = CHART_CONFIG_VERSION
+): any => {
+  const fromVersion = config.version || "1.0.0";
+  const direction = upOrDown(fromVersion, toVersion);
+
+  if (direction === "same") {
+    return config;
+  }
+
+  const currentMigration = migrations.find(
+    (migration) => migration[direction === "up" ? "from" : "to"] === fromVersion
+  );
+
+  if (currentMigration) {
+    const newConfig = currentMigration[direction](config);
+    return migrateChartConfig(newConfig, toVersion);
```

Comment:
Took me some time to understand how we would go to the latest version, I was searching for a `while toVersion !== curVersion`, but now I noticed the recursive call, guess I am tired 😅